javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/DefineClassHandler.java

```diff
@@ -63,33 +93,37 @@ public void afterDefineClass(DefineClassContext context) {
   /**
    * Detect whether loading the specified class is known to fail.
    *
-   * @param className class being loaded
+   * @param dotClassName class being loaded
    * @return true if it is known that loading class with given name will fail
    */
-  public static boolean isFailedClass(String className) {
+  public static boolean isFailedClass(String dotClassName) {
     DefineClassContextImpl context = defineClassContext.get();
-    return context.failedClassName != null && context.failedClassName.equals(className);
+    return context.failedClassDotName != null && context.failedClassDotName.equals(dotClassName);
+  }
+
+  public static Set<String> getSuperTypes() {
+    Set<String> superNames = defineClassContext.get().superDotNames;
+    return superNames == null ? Collections.emptySet() : superNames;
   }
 
   private static class DefineClassContextImpl implements DefineClassContext {
     private static final DefineClassContextImpl NOP = new DefineClassContextImpl();
```

Comment:
I tried it and unfortunately it didn't work well because we keep context in `ThreadLocal<DefineClassContextImpl>` if it was changed to `DefineClassContext` we'd need to add casts or add more things to `DefineClassContext` interface.